### PR TITLE
[nan-526] for OAauth2 force an expiration unless explicitly stated

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
A customer imported connections and didn't import the expiration date. Then one day the connections stopped working because the system never know when to refresh their access token. Unless a provider provides tokens that never expire force on import an expires_at or expires_in parameter

## Issue ticket number and link
NAN-526

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

**Comment by TBonnin** [comment]
what does importing a connection means? using the api?

**Comment by khaliqgant** [comment]
> what does importing a connection means? using the api?

Yeah exactly: http://localhost:3033/reference/api/connection/post

File: packages/server/lib/controllers/connection.controller.ts [hunk 1]
```diff
@@ -534,6 +534,12 @@ class ConnectionController {
                     errorManager.errRes(res, 'missing_access_token');
                     return;
                 }
+
+                if (!parsedExpiresAt && noExpiration && noExpiration !== true) {
```
Comment (TBonnin) [inline packages/server/lib/controllers/connection.controller.ts]: why the `&& noExpiration`? isn't the last part enough?

**Review by TBonnin — APPROVED** [review]

File: packages/server/lib/controllers/connection.controller.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/server/lib/controllers/connection.controller.ts]: Thanks, removed with https://github.com/NangoHQ/nango/pull/1827/commits/964a99c105175d9931dc57d9daf042728cade61e

**Pull request merged**